src/Math-Matrix/PMMatrix.class.st

```diff
@@ -824,6 +824,24 @@ PMMatrix >> qrFactorizationWithPivoting [
 	^ Array with: q with: r with: pivot
 ]
 
+{ #category : #operation }
+PMMatrix >> raisedTo: aPower [
+
+	" Answers the receiver raised to a power, aPower .
+	 If aPower is negative, inverse of the receiver is raised to the absolute value of aPower."
+
+	self assert: self isSquare description: 'Matrix should be square'.
+	
+	 aPower = 0 ifTrue: [ 
+		^ PMMatrix identity: self numberOfRows ].
+	
+	aPower > 0 ifTrue: [
+		^ self * (self raisedTo: aPower -1) ].
```

Comment:
Recursion! Is there a way to iterate instead, make it faster?

Reply:
@hemalvarambhia Yes, I could change the recursion to iteration. I had kept recursion because I felt that the code looked more concise and elegant.